subcommands/keys/tuf_updates_rotate_offline_key.go

```diff
@@ -186,19 +186,16 @@ func doTufUpdatesRotateOfflineTargetsKey(cmd *cobra.Command) {
 		subcommands.DieNotNil(err)
 	}
 
-	targetsMap, err := api.ProdTargetsList(factory, false)
+	targetsProdMap, err := api.ProdTargetsList(factory, false)
 	subcommands.DieNotNil(err, "Failed to fetch production targets:")
-outerLoop:
-	for tag, targets := range targetsMap {
-		for _, sig := range targets.Signatures {
-			if sig.KeyID == oldestKey.Id {
-				continue outerLoop
-			}
-		}
-		// These targets does not contain a signature by a rotated key - skip them
-		delete(targetsMap, tag)
-	}
-	newTargetsSigs, err := signProdTargets(newKey, targetsMap)
+	excludeTargetsWithoutKeySigInplace(targetsProdMap, oldestKey.Id)
+	newTargetsProdSigs, err := signProdTargets(newKey, targetsProdMap)
+	subcommands.DieNotNil(err)
+
+	targetsWaveMap, err := api.WaveTargetsList(factory, false)
+	subcommands.DieNotNil(err, "Failed to fetch production wave targets:")
+	excludeTargetsWithoutKeySigInplace(targetsWaveMap, oldestKey.Id)
```

Comment:
I am just wondering if it's possible that some old wave's targets wasn't re-signed after targets key rotation (because we didn't support it before). So, if we filter in by ` oldestKey.Id` it means that such `targetsWave` won't be re-signed.

Reply:
but devices will reject such wave targets anyway.